pkg/controllers/user/logging/utils/embedded.go

```diff
@@ -201,6 +210,120 @@ func RemoveEmbeddedTarget(dep rv1beta2.DeploymentInterface, sa rv1.ServiceAccoun
 	return nil
 }
 
+func UpdateEmbeddedEndpoint(podLister rv1.PodLister, serviceLister rv1.ServiceLister, clusterLoggings v3.ClusterLoggingInterface) {
+	timeout := time.After(2 * time.Minute)
+	tick := time.Tick(10 * time.Second)
+	for {
+		select {
+		case <-timeout:
+			logrus.Warn("update embedded endpoint timeout")
+			return
+		case <-tick:
+			cls, err := clusterLoggings.List(metav1.ListOptions{})
```

Comment:
Can we use cache here

Reply:
because it will be update below, so I don't use cache here.

Reply:
Alena suggested we use cache even for update case
>one thing regarding an update. It is ok to get object-to-update from cache. It is also fine if the update would fail the first time due to object changed in API. It will be subsequently retried.